src/Microsoft.DotNet.Darc/src/DarcLib/RemoteRepoBase.cs

```diff
@@ -45,81 +45,46 @@ protected RemoteRepoBase(string temporaryRepositoryPath)
+                    string clonedRepo = null;
 
-                    using (LibGit2Sharp.Repository localRepo = new LibGit2Sharp.Repository(repoPath))
+                    using (_logger.BeginScope("Sparse and shallow checkout of branch {branch} in {repoUri}...", branch, repoUri))
                     {
-                        foreach (GitFile file in filesToCommit)
-                        {
-                            string filePath = Path.Combine(tempRepoFolder, file.FilePath);
-
-                            if (file.Operation == GitFileOperation.Add)
-                            {
-                                if (!File.Exists(filePath))
-                                {
-                                    string parentFolder = Directory.GetParent(filePath).FullName;
+                        clonedRepo = LocalHelpers.SparseAndShallowCheckout(repoUri, branch, tempRepoFolder, _logger, remote, dotnetMaestro, pat);
+                    }
 
-                                    Directory.CreateDirectory(parentFolder);
-                                }
+                    foreach (GitFile file in filesToCommit)
+                    {
+                        string filePath = Path.Combine(clonedRepo, file.FilePath);
 
-                                using (FileStream stream = File.Create(filePath))
-                                {
-                                    byte[] contentBytes = GetUtf8ContentBytes(file.Content, file.ContentEncoding);
-                                    await stream.WriteAsync(contentBytes, 0, contentBytes.Length);
-                                }
+                        if (file.Operation == GitFileOperation.Add)
+                        {
+                            if (!File.Exists(filePath))
+                            {
+                                string parentFolder = Directory.GetParent(filePath).FullName;
 
-                                LibGit2SharpHelpers.AddFileToIndex(localRepo, file, filePath, _logger);
+                                Directory.CreateDirectory(parentFolder);
                             }
-                            else
+
+                            using (FileStream stream = File.Create(filePath))
                             {
-                                File.Delete(Path.Combine(tempRepoFolder, file.FilePath));
+                                byte[] contentBytes = GetUtf8ContentBytes(file.Content, file.ContentEncoding);
+                                await stream.WriteAsync(contentBytes, 0, contentBytes.Length);
                             }
                         }
-
-                        localRepo.Index.Write();
-
-                        LibGit2Sharp.Signature author = new LibGit2Sharp.Signature(dotnetMaestro, $"@{dotnetMaestro}", DateTime.Now);
-                        LibGit2Sharp.Signature commiter = author;
-                        localRepo.Commit(commitMessage, author, commiter, new LibGit2Sharp.CommitOptions
+                        else
```

Comment:
NIT: this seems a risky operation.. wouldn't it be better to explicit compare if it's indeed a delete operation just to make sure we don't end up deleting something in GitFileOperation someday gets other operations added to it.